lib/ld-eventsource/client.rb

```diff
@@ -328,7 +287,24 @@ def read_stream(cxn)
       # it can automatically reset itself if enough time passes between failures.
       @backoff.mark_success
 
-      event_parser = Impl::EventParser.new(read_lines)
+      chunks = Enumerator.new do |gen|
```

Comment:
This is what was previously in `read_chunk_into_buffer` above. `BufferedLineReader` takes an enumerator of data chunks as a parameter, so that it can process the data abstractly without caring about whether it's from an HTTP connection or whatever. Here's the implementation of that enumerator that specifically reads from a connection.

Reply:
Interestingly `HTTP.rb` has a response body (`HTTP::Response::Body#each`) which you can treat as an `Enumerable` entity:

https://github.com/httprb/http/blob/main/lib/http/response/body.rb#L36

```ruby
# Iterate over the body, allowing it to be enumerable
def each
  while (chunk = readpartial)
    yield chunk
  end
end
```

...which may reduce the amount of code necessary to work with  chunks down to something like:

```ruby
chunks = Enumerator.new do |gen|
  begin
    cxn.each do |chunk|
      break if @stopped.value
      break if chunk.nil?

      gen.yield chunk
    end
  rescue HTTP::TimeoutError 
    # For historical reasons, we rethrow this as our own type
    raise Errors::ReadTimeoutError.new(@read_timeout)
  end
end
```

...though I really wish they'd approve and merge the block-scoped rescues so you could do this instead:

```ruby
chunks = Enumerator.new do |gen|
  cxn.each do |chunk|
    break if @stopped.value
    break if chunk.nil?

    gen.yield chunk
  rescue HTTP::TimeoutError 
    # For historical reasons, we rethrow this as our own type
    raise Errors::ReadTimeoutError.new(@read_timeout)
  end
end
```